lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -112,17 +121,64 @@ def update_status(self, message):
         return self.task_progress.update_task_state(extra_meta={'step': message})
 
 
+class _CertificateBulkContext(object):
+    def __init__(self, context, users):
+        certificate_whitelist = CertificateWhitelist.objects.filter(course_id=context.course_id, whitelist=True)
+        self.whitelisted_user_ids = [entry.user_id for entry in certificate_whitelist]
+        self.certificates_by_user = {
+            certificate.user.id: certificate
+            for certificate in
+            GeneratedCertificate.objects.filter(course_id=context.course_id, user__in=users)
+        }
+
+
+class _TeamBulkContext(object):
+    def __init__(self, context, users):
+        if context.teams_enabled:
+            self.teams_by_user = {
+                membership.user.id: membership.team.name
+                for membership in
+                CourseTeamMembership.objects.filter(team__course_id=context.course_id, user__in=users)
+            }
+        else:
+            self.teams_by_user = {}
+
+
+class _EnrollmentBulkContext(object):
+    def __init__(self, context, users):
+        CourseEnrollment.bulk_fetch_enrollment_states(users, context.course_id)
+        self.verified_users = [
+            verified.user.id for verified in
+            SoftwareSecurePhotoVerification.verified_query().filter(user__in=users).select_related('user__id')
+        ]
+
+
+class _CourseGradeBulkContext(object):
+    def __init__(self, context, users):
+        self.certs = _CertificateBulkContext(context, users)
+        self.teams = _TeamBulkContext(context, users)
+        self.enrollments = _EnrollmentBulkContext(context, users)
+        bulk_cache_cohorts(context.course_id, users)
+        BulkRoleCache.prefetch(users)
+        PersistentCourseGrade.prefetch(context.course_id, users)
+        BulkCourseTags.prefetch(context.course_id, users)
+
+
 class CourseGradeReport(object):
     """
     Class to encapsulate functionality related to generating Grade Reports.
     """
+    # Batch size for chunking the list of enrollees in the course.
+    USER_BATCH_SIZE = 100
```

Comment:
Nah, definitely not a blocker, just wanted to raise the question. Without already having a django admin view and with new releases going out daily, I'm okay to define the batch size this way.